Rename --keepEmptyParagraphs to --keepEmptySections

ziaddevv opened the pull request:
- Updated cli parameter from --keepEmptyParagraphs to --keepEmptySections
- Updated all internal references in TypeScript source files

Fixes openzim/mwoffliner#2513

kelson42 commented:
We will have to wait the release of version 1.17.0 before merging this PR.

benoit74 commented on src/renderers/abstract.renderer.ts:
```diff
@@ -768,7 +768,7 @@ export abstract class Renderer {
     })
 
     /* Remove empty paragraphs */
```
Please fix comment as well

benoit74 commented:
@ziaddevv please also add an entry in `CHANGELOG` file

ziaddevv commented:
> @ziaddevv please also add an entry in `CHANGELOG` file

Thank you for the feedback! I've updated the comment and added an entry in the ChangeLog file

benoit74 reviewed (approved):
LGTM. Moving to draft to not merge it by accident, to be merged only at next major version

Markus-Rost commented on test/unit/saveArticles.test.ts:
```diff
@@ -199,7 +199,7 @@ describe('saveArticles', () => {
 
     test(`Keep empty paragraphs for ${renderer} renderer`, async () => {
```
Please update the description of the test as well.

ziaddevv replied:
done , so the test just right above it have the same issue but it doesn't directly reference the new CLI behavior (keepEmptySections)  do i have to change it ?  test(`Remove empty paragraphs for ${renderer} renderer`, async () => {

ziaddevv replied:
also what about the old changelogs ?

kelson42 commented:
> I kept the parameter descriptions the same so that users running --help will still see the deprecated options, making the backward compatibility clear , Is that ok ?

No, please move everything to the new naming. Just secure that if someone use the old option naming, then it continues to work fine.

ziaddevv commented:
no i just meant to say that in this step  `Have both names in the parameter list, description of keepEmptyParagraphs suggesting the new name ` I kept the description of keepEmptyParagraphs the same as KeepEmptySections and user can see the suggestion say ` [deprecated: use --keepEmptySections instead]`
